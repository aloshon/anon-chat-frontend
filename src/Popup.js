import React from "react";
import "./Popup.css";

/** 
 * Component that takes content and a handleClose function to give
 * purpose to the close icon.
 * Content can be text or a list.
*/  

const Popup = ({content, handleClose}) => {
    return (
        <div className="popup-box">
            <div className="box">
                <span className="close-icon" onClick={handleClose}>X</span>
                {content}
            </div>
        </div>
    )
}

export default Popup;