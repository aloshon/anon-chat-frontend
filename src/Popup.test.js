import { render, screen } from '@testing-library/react';
import Popup from './Popup';


describe("Popup should render content and button", () => {
    const content = "this is the content";
    const handleClose = () => undefined;

    const {asFragment} = render(
        <Popup content={content} handleClose={handleClose}/>
    );

    test("renders popup with content", () => {
        const content = screen.getByText("this is the content");
        expect(content).toBeInTheDocument();
    });
  
    test("matches snapshot", () => {
      expect(asFragment()).toMatchSnapshot();
    });
    
});