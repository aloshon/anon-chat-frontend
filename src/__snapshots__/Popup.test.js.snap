// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup should render content and button matches snapshot 1`] = `<DocumentFragment />`;
